OWML.ModHelper.Menus/ModInputCombinationPopup.cs

```diff
@@ -0,0 +1,201 @@
+﻿using OWML.Common;
+using OWML.Common.Menus;
+using OWML.ModHelper.Input;
+using OWML.ModHelper.Events;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using System;
+using System.Collections.ObjectModel;
+using System.Reflection;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputCombinationPopup : PopupMenu
+    {
+        public ILayoutManager Layout { get; private set; }
+        public event Action OnPopupReset;
+        public string Combination
+        {
+            get
+            {
+                string result = "";
+                for (var j = 0; j < _combination.Count; j++)
+                {
+                    result += ModInputLibrary.KeyCodeToString(_combination[j]);
+                    if (j < _combination.Count - 1)
+                    {
+                        result += "+";
+                    }
+                }
+                return result;
+            }
+        }
+        public ReadOnlyCollection<KeyCode> KeyCodes => _combination.AsReadOnly();
+
+        private SubmitAction _resetAction;
+        private ButtonWithHotkeyImageElement _resetButton;
+        private List<KeyCode> _combination = new List<KeyCode>();
+        private IModInputHandler _inputHandler;
+        private bool _wasReleased = true;
+        protected SingleAxisCommand _resetCommand;
+
+        protected override void InitializeMenu()
+        {
+            base.InitializeMenu();
+            if (this._resetAction != null)
+            {
+                this._resetAction.OnSubmitAction += this.InvokeReset;
+            }
```

Comment:
`this._resetAction?.OnSubmitAction += this.InvokeReset;`

Reply:
I think I tried that and you can't use '?' with assign-like operations

Reply:
Huh, why did it get marked as resolved